components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -100,8 +111,9 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
 
   ngOnInit(): void {
     this.ruleForm = this.fb.group({
-      name: [this.rule.name || '', Validators.required],
-      type: [this.rule.type || '', Validators.required],
+      // Must stay in sync with error checks in project-rules.component.html
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      type: [{ value: this.rule.type || '', disabled: this.editingRule } , Validators.required],
```

Comment:
This little gem had not been used in the code base before AFAIK. Can't take the credit, though: Angular helped me find what to do. Initially I just put `[disabled] = "editingRule"` in the template but at runtime Angular patiently explained that could cause "changed after checked" errors and should use this pattern instead. This initializes the element with the `disabled` property given here.